app/api/suggestions/specs/routes.spec.ts

```diff
@@ -0,0 +1,260 @@
+/* eslint-disable max-lines */
+import request from 'supertest';
+import { Application, NextFunction, Request, Response } from 'express';
+import entities from 'api/entities';
+
+import { search } from 'api/search';
+import { WithId } from 'api/odm';
+import { suggestionsRoutes } from 'api/suggestions/routes';
+import { testingEnvironment } from 'api/utils/testingEnvironment';
+import {
+  fixtures,
+  shared2enId,
+  shared6enId,
+  suggestionSharedId6Enemy,
+  suggestionSharedId6Title,
+} from 'api/suggestions/specs/fixtures';
+import { setUpApp } from 'api/utils/testingRoutes';
+import { SuggestionState } from 'shared/types/suggestionSchema';
+import { EntitySchema } from 'shared/types/entityType';
+
+jest.mock(
+  '../../utils/languageMiddleware.ts',
+  () => (req: Request, _res: Response, next: NextFunction) => {
+    req.language = 'en';
+    next();
+  }
+);
+
+jest.mock('api/services/informationextraction/InformationExtraction', () => ({
+  InformationExtraction: class IXMock {
+    status = jest.fn().mockResolvedValue({ status: 'ready' });
+
+    trainModel = jest.fn().mockResolvedValue({ status: 'processing' });
+  },
+}));
+
+describe('suggestions routes', () => {
+  let user: { username: string; role: string } | undefined;
+  const getUser = () => user;
+
+  beforeAll(async () => {
+    await testingEnvironment.setUp(fixtures);
+  });
+  beforeEach(async () => {
+    user = { username: 'user 1', role: 'admin' };
+    spyOn(search, 'indexEntities').and.returnValue(Promise.resolve());
+  });
+
+  const app: Application = setUpApp(
+    suggestionsRoutes,
+    (req: Request, _res: Response, next: NextFunction) => {
+      (req as any).user = getUser();
+      next();
+    }
+  );
+
+  afterAll(async () => testingEnvironment.tearDown());
+
+  describe('GET /api/suggestions', () => {
+    it('should return the suggestions filtered by the request language and the property name', async () => {
+      const response = await request(app)
+        .get('/api/suggestions')
+        .query({ filter: { propertyName: 'super_powers' } })
+        .expect(200);
+      expect(response.body.suggestions).toMatchObject([
```

Comment:
How does the "filtered by language work"? We mock an English request, but we still expect the Spanish entity returned as well?

Reply:
The returned is actually the title from the English entity, but the language of the suggestion is in Spanish, we are not filtering the suggestions by their language but by the language of entities. I added the language to the title to make it clear.